src/controllers/expensesController.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const expensesService = require('../services/expensesServices');
+const usersService = require('../services/usersServices');
+
+beforeEach(() => {
+  expensesService.deleteAllExpenses();
+});
+
+const getExpenses = (req, res) => {
+  const {
+    userId,
+    from,
+    to,
+    categories,
+  } = req.query;
+
+  let expenses = expensesService.getExpenses();
+
+  if (userId) {
+    expenses = expenses.filter(expense => expense.userId === +userId);
+  }
+
+  if (from) {
+    expenses = expenses.filter(
+      expense => new Date(expense.spentAt) > new Date(from)
+    );
+  }
+
+  if (to) {
+    expenses = expenses.filter(
+      expense => new Date(expense.spentAt) < new Date(to)
+    );
+  }
+
+  if (categories) {
+    expenses = expenses.filter(
+      expense => categories.includes(expense.category)
```

Comment:
if there are no `categories` in search params we will get `undefined`
if only 1 `categories` we will get just value (....?categories=1)
And if we have many categories we will get array of values (....?categories=1&categories=2&categories=3)

![image](https://github.com/mate-academy/node_accounting-app-v2/assets/86789161/ea010564-92a2-4116-9fcf-a66739d6d01f)

You code will crash if you have only 1 category